tests/unit_tests/test_dynamic_prompt.py

```diff
@@ -0,0 +1,93 @@
+"""Test functionality related to dynamic prompts."""
+from langchain.prompt import DynamicPrompt, Prompt
+
+
+def test_dynamic_prompt_valid() -> None:
+    """Test dynamic prompt can be successfully constructed from examples."""
+    template = """Test Prompt:
+
+Question: who are you?
+Answer: foo
+
+Question: what are you?
+Answer: bar
+
+Question: {question}
+Answer:"""
+    input_variables = ["question"]
+    example_separator = "\n\n"
+    prefix = """Test Prompt:"""
+    suffix = """Question: {question}\nAnswer:"""
+    examples = [
+        """Question: who are you?\nAnswer: foo""",
+        """Question: what are you?\nAnswer: bar""",
+    ]
+    dynamic_prompt = DynamicPrompt(
+        examples=examples,
+        suffix=suffix,
+        input_variables=input_variables,
+        example_separator=example_separator,
+        prefix=prefix,
+    )
+    prompt_from_template = Prompt(input_variables=input_variables, template=template)
+    assert dynamic_prompt.format(question="foo?") == prompt_from_template.format(
+        question="foo?"
+    )
+    assert dynamic_prompt.input_variables == prompt_from_template.input_variables
+
+
+def test_dynamic_prompt_trims_examples() -> None:
+    """Test dynamic prompt can be successfully constructed from examples."""
+    longer_template = """Test Prompt:
```

Comment:
this is a complete nit but i hate how this looks visually lol. i ran into this on prev prs - can we either put the prompt outside the test function OR do something like 
```
    shortest_template = (
"Test Prompt:\n"
"Question: {question}\n"
Answer:"
)
```
or is there a better way to do it? to be clear the part i dont like is the weird indent

Reply:
yes i agree! reformatted